File: baselibrary/src/main/java/com/anningtex/baselibrary/base/AbsBaseActivity.java
package com.anningtex.baselibrary.base;

import android.content.Context;
import android.content.Intent;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.os.Build;
import android.os.Bundle;
import android.os.LocaleList;
import android.util.DisplayMetrics;
import android.view.WindowManager;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.anningtex.baselibrary.manager.AppManager;

import java.util.Locale;

public abstract class AbsBaseActivity<P extends BaseContract.Presenter> extends AppCompatActivity implements BaseContract.View {
    protected P mPersenter;

    protected WindowManager mWindowManager;
    protected DisplayMetrics mMetrics;
    protected Bundle mBundle;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutResId());
        setLangue();
        mWindowManager = (WindowManager) getSystemService(Context.WINDOW_SERVICE);
        mMetrics = new DisplayMetrics();
        mWindowManager.getDefaultDisplay().getMetrics(mMetrics);
        mBundle = getIntent().getExtras();
        AppManager.getAppManager().addActivity(this);
        initViews();
    }

    protected abstract int getLayoutResId();

    /**
     * 初始化控件，设置控件事件
     */
    protected abstract void initViews();

    public void $startActivity(Class<?> claz) {
        Intent intent = new Intent(this, claz);
        startActivity(intent);
    }

    public void $startActivity(Class<?> cls, String name, String value) {
        Intent intent = new Intent(this, cls);
        intent.putExtra(name, value);
        startActivity(intent);
    }

    public void $startActivity(Class<?> cls, Bundle bundle) {
        Intent intent = new Intent(this, cls);
        intent.putExtras(bundle);
        startActivity(intent);
    }

    @Override
    public void toast(String msg) {
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show();
    }

    @Override
    public void toast(int msg) {
        Toast.makeText(this, getString(msg), Toast.LENGTH_SHORT).show();
    }

    /**
     * 跟随手机系统判断转换中英文
     */
    public void setLangue() {
        Locale locale;
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            locale = LocaleList.getDefault().get(0);
        } else {
            locale = Locale.getDefault();
        }
        String language = locale.getLanguage();
        Resources resources = getResources();
        Configuration config = resources.getConfiguration();
        DisplayMetrics dm = resources.getDisplayMetrics();
        if (language.equals("zh")) {
            config.locale = Locale.SIMPLIFIED_CHINESE;
        } else {
            config.locale = Locale.ENGLISH;
        }
        resources.updateConfiguration(config, dm);
    }

    @Override
    public void showLoading() {

    }

    @Override
    public void dismissLoading() {

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (mPersenter != null) {
            mPersenter.detachView();
        }
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        AppManager.getAppManager().finishCurrentActivity();
    }
}

File: baselibrary/src/main/java/com/anningtex/baselibrary/base/AbsBaseActivity.kt
package com.anningtex.baselibrary.base

import android.content.Context
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.os.LocaleList
import android.util.DisplayMetrics
import android.view.WindowManager
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.anningtex.baselibrary.manager.AppManager
import java.util.*

abstract class AbsBaseActivity<P : AbsBaseContract.Presenter<*>?> : AppCompatActivity(), AbsBaseContract.View {
    @JvmField
    protected var mPresenter: P? = null
    protected var mWindowManager: WindowManager? = null

    @JvmField
    protected var mMetrics: DisplayMetrics? = null
    protected var mBundle: Bundle? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutResId)
        setLanguage()
        mWindowManager = getSystemService(Context.WINDOW_SERVICE) as WindowManager
        mMetrics = DisplayMetrics()
        mWindowManager!!.defaultDisplay.getMetrics(mMetrics)
        mBundle = intent.extras
        AppManager.getAppManager().addActivity(this)
        initViews()
    }

    protected abstract val layoutResId: Int

    /**
     * 初始化控件，设置控件事件
     */
    protected abstract fun initViews()

    fun startAct(claz: Class<*>?) {
        val intent = Intent(this, claz)
        startActivity(intent)
    }

    fun startAct(cls: Class<*>?, name: String?, value: String?) {
        val intent = Intent(this, cls)
        intent.putExtra(name, value)
        startActivity(intent)
    }

    fun startAct(cls: Class<*>?, bundle: Bundle?) {
        val intent = Intent(this, cls)
        intent.putExtras(bundle!!)
        startActivity(intent)
    }

    override fun toast(msg: String) {
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show()
    }

    override fun toast(msg: Int) {
        Toast.makeText(this, getString(msg), Toast.LENGTH_SHORT).show()
    }

    /**
     * 跟随手机系统判断转换中英文
     */
    fun setLanguage() {
        val locale: Locale
        locale = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            LocaleList.getDefault()[0]
        } else {
            Locale.getDefault()
        }
        val language = locale.language
        val resources = resources
        val config = resources.configuration
        val dm = resources.displayMetrics
        if (language == "zh") {
            config.locale = Locale.SIMPLIFIED_CHINESE
        } else {
            config.locale = Locale.ENGLISH
        }
        resources.updateConfiguration(config, dm)
    }

    override fun showLoading() {}
    override fun dismissLoading() {}
    override fun onDestroy() {
        super.onDestroy()
        mPresenter?.detachView()
    }

    override fun onBackPressed() {
        super.onBackPressed()
        AppManager.getAppManager().finishCurrentActivity()
    }
}